internal/characters/alhaitham/cons.go

```diff
@@ -0,0 +1,110 @@
+// max 4 stacks.
+// Each stack's duration is counted independently.
+// This effect can be triggered even when the maximum number of Chisel-Light Mirrors has been reached.
+func (c *char) c2() {
+	m := make([]float64, attributes.EndStatType)
+	m[attributes.EM] = 50
+	c.AddStatMod(character.StatMod{
+		Base:         modifier.NewBaseWithHitlag(fmt.Sprintf("alhaitham-c2-%v-stack", c.c2counter+1), 480), //8s
+		AffectedStat: attributes.EM,
+		Amount: func() ([]float64, bool) {
+
+			return m, true
+		},
+	})
+	c.c2counter = (c.c2counter + 1) % 4 //stacks are independent from each other, this will cycle them
+}
+
+// When Particular Field: Fetters of Phenomena is unleashed, the following effects will become active
+// based on the number of Chisel-Light Mirrors consumed and created this time around:
+// ·Each Mirror consumed will increase the Elemental Mastery of all other nearby party members by 30 for 15s.
+// ·Each Mirror generated will grant Alhaitham a 10% Dendro DMG Bonus for 15s.
+// The pre-existing duration of the aforementioned effects will be cleared if you use Particular Field: Fetters of Phenomena again while they are in effect
+func (c *char) c4(src string, counter int) {
+	//TODO: Does C4 apply the dendro dmg bonus to alhaitham even if he is NOT on field during the refund?
+	m := make([]float64, attributes.EndStatType)
+	if src == "gain" {
+		m[attributes.DendroP] = 0.1
+		c.AddStatMod(character.StatMod{
+			Base:         modifier.NewBaseWithHitlag(fmt.Sprintf("alhaitham-c4-%v-instance", counter), 900), //15s
+			AffectedStat: attributes.DendroP,
+			Amount: func() ([]float64, bool) {
+
+				return m, true
+			},
+		})
+		c.Core.Log.NewEvent("c4 buff on gain", glog.LogCharacterEvent, c.Index)
+	} else {
+		m[attributes.EM] = 30
+		for i, char := range c.Core.Player.Chars() {
+			// skip Alhaitham
+			if i == c.Index {
+				continue
+			}
+			char.AddStatMod(character.StatMod{
+				Base:         modifier.NewBaseWithHitlag(fmt.Sprintf("alhaitham-c4-%v-instance", counter+1), 900),
+				AffectedStat: attributes.EM,
+				Amount: func() ([]float64, bool) {
+					return m, true
+				},
+			})
+		}
+		c.Core.Log.NewEvent("c4 buff on loss", glog.LogCharacterEvent, c.Index)
+	}
+
+}
+
+// Alhaitham gains the following effects:
+// · 2 seconds after Particular Field: Fetters of Phenomena is unleashed,
+// he will generate 3 Chisel-Light Mirrors regardless of the number of mirrors consumed.
+//
+// · If Alhaitham generates Chisel-Light Mirrors when their numbers have already maxed out,
+// his CRIT Rate and CRIT DMG will increase by 10% and 70% respectively for 6s.
+// If this effect is triggered again during its initial duration, the duration remaining will be increased by 6s.
+const c6key = "alhaitham-c6"
+
+func (c *char) c6() {
+	m := make([]float64, attributes.EndStatType)
+	m[attributes.CR] = 0.1
+	m[attributes.CD] = 0.7
+
+	if c.StatModIsActive(c6key) { //TODO: Needs a check to work only during the first 6s of the buff?
```

Comment:
No, I don't think it needs a check like that. Checking if C6 is already up is correct.